src/AST/Names.dfy

```diff
@@ -28,6 +29,30 @@ module {:options "-functionSyntax:4"} Bootstrap.AST.Names {
         case Name(parent, suffix) => parent.ToString() + "." + suffix
     }
 
+    function ToDafnyName(): string {
+      var parts := Seq.Filter(this.ToSeq(), a => a !in {"_default", "_module"});
+      Seq.Flatten(Seq.Interleave(".", parts))
+    }
+
+    function IsInternal(): bool {
+      match this {
+        case Anonymous => false
+        case Name(Anonymous, "_System") => true
+        case Name(parent, suffix) =>
+          var parts := Seq.Split('_', suffix);
+          (|parts| > 0 && parts[0] == "reveal") || parent.IsInternal()
+      }
+    }
+
+    function IsCompile(): bool {
+      match this {
+        case Anonymous => false
+        case Name(_, suffix) =>
+          var parts := Seq.Split('_', suffix);
+          (|parts| > 0 && parts[|parts| - 1] == "Compile") || parent.IsCompile()
```

Comment:
```suggestion
          "Compile_" <= suffix || parent.IsCompile()
```

Reply:
Maybe this could be generalized to a predicate "Any" on names, and then it can be inlined: `name.Any(s => "Compile_" <= s)`

Same for internal above.